# **Data Inference**

## **I. Import Library**

In [15]:
import pandas as pd
import numpy as np
import pickle

## **II. Create Dummy Data**

In [20]:
np.random.seed(42)

data_dummy = pd.DataFrame({
    'pm10': np.random.uniform(10, 150, size=10),
    'pm25': np.random.uniform(5, 100, size=10),
    'so2': np.random.uniform(1, 10, size=10),
    'co': np.random.uniform(1, 100, size=10),
    'o3': np.random.uniform(1, 100, size=10),
    'no2': np.random.uniform(1, 50, size=10),
    'max': np.random.uniform(20, 100, size=10),  
    'station': ['DKI1 (Bunderan HI)'] * 10,       
    'critical': ['CO'] * 10,
    'category': ['SEDANG'] * 10             
})


print("Data Dummy untuk Inference:")
print(data_dummy)

Data Dummy untuk Inference:
         pm10       pm25       so2         co         o3        no2  \
0   62.435617   6.955527  6.506676  61.146940  13.081785  48.509647   
1  143.100003  97.141436  2.255445  17.881888  50.022514  38.981508   
2  112.479152  84.082051  3.629302   7.440108   4.404464  47.035448   
3   93.812188  25.172216  4.297257  94.939668  91.022720  44.846540   
4   31.842610  22.273372  5.104630  96.597571  26.619218  30.297099   
5   31.839233  22.423428  8.066584  81.031337  66.589706  46.171838   
6   18.131706  33.903013  2.797064  31.156763  31.859397   5.336133   
7  131.264660  54.851861  5.628110  10.669539  52.486734  10.603160   
8   94.156102  46.034777  6.331731  68.739070  55.124318   3.216137   
9  109.130161  32.666768  1.418054  44.575097  19.300591  16.941186   

         max             station critical category  
0  51.094183  DKI1 (Bunderan HI)       CO   SEDANG  
1  41.707923  DKI1 (Bunderan HI)       CO   SEDANG  
2  86.299001  DKI1 (Bunderan HI

## **III. Save the Dummy Data**

In [21]:
data_dummy.to_csv('data_dummy_aqi_inference.csv', index=False)

## **IV. Load Model**

In [22]:
with open('best_model_rf.pkl', 'rb') as file:
    best_model_rf = pickle.load(file)

## **V. Prediction**

In [23]:
predictions = best_model_rf.predict(data_dummy)

print("\nPrediksi Air Quality Index (AQI) pada Data Dummy:")
print(predictions)


Prediksi Air Quality Index (AQI) pada Data Dummy:
[93.695 88.715 91.365 94.885 94.    93.375 94.    93.01  94.    94.   ]


- ``93.695:`` AQI prediksi untuk baris pertama data dummy.
- ``88.715:`` AQI prediksi untuk baris kedua data dummy.
- ...dan seterusnya.